src/pages/collab/index.tsx

```diff
@@ -0,0 +1,287 @@
+
+    const value = event.currentTarget.getAttribute("value");
+
+    setPageState((prev) => ({
+      ...prev,
+      difficulty: parseInt(value!),
+    }));
+
+    if (pageState.isDifficultyMissing) {
+      setPageState((prev) => ({
+        ...prev,
+        isDifficultyMissing: false,
+      }));
+    }
+
+    const displayedDifficulty = document.querySelector(
+      ".difficulty-menu .select",
+    );
+    if (displayedDifficulty)
+      displayedDifficulty.innerHTML = event.currentTarget.innerHTML;
+  };
+
+  const onCategoryChange = (value: string) => {
+    if (!pageState.hasSetCategory)
+      setPageState((prev) => ({ ...prev, hasSetCategory: true }));
+
+    setPageState((prev) => ({
+      ...prev,
+      category: value,
+      isCategoryMissing: value ? false : true,
+    }));
+  };
+
+  // TODO: join session
+  const joinSession = () => {
+    console.log("join session");
+  };
+
+  return (
+    <div className="min-h-screen bg-[#2f2f2f] text-center">
+      {pageState.isWaiting && (
+        <div className="queue-status-bar absolute left-1/2 w-80 -translate-x-1/2 justify-self-center rounded-md bg-black text-white">
+          {pageState.isTimerActive && <LoadingIcon />}
+          <span>{pageState.statusMessage}</span>
+          {pageState.isTimerActive && (
+            <button onClick={cancelRequest} className="h-6 w-6 rounded-full">
+              <FontAwesomeIcon
+                icon={faXmark}
+                size="lg"
+                style={{ color: "#ff0000" }}
+              />
+            </button>
+          )}
+          {!pageState.isTimerActive && pageState.requestFailed && (
+            <button
+              onClick={addRequest}
+              className="h-6 w-12 rounded-full bg-gray-500 text-white"
+            >
+              Retry
+            </button>
+          )}
+          {!pageState.isTimerActive && !pageState.requestFailed && (
+            <button
+              onClick={joinSession}
+              className="h-6 w-24 rounded-md bg-green-500 text-white"
+            >
+              Join Room
+            </button>
+          )}
+        </div>
+      )}
+      {pageState.isTimerActive && (
+        <div className="absolute left-1/2 top-9 w-1/6 -translate-x-1/2 justify-self-center rounded-md bg-black py-1 text-white">
+          Time elapsed: {pageState.waitingTime} seconds
+        </div>
+      )}
+      <div className="absolute left-1/2 top-1/2 flex h-80 w-64 -translate-x-1/2 -translate-y-1/2 flex-col justify-evenly rounded-xl border p-5">
+        <div className="difficulty-menu">
+          <span className="choose">Choose Difficulty</span>
+
+          <div className="dropdown" onClick={onDifficultySelect}>
+            <div className="select text-center">
+              <span>Select Difficulty</span>
+            </div>
+            <ul className="dropdown-menu">
+              <li value={0} onClick={(event) => selectDifficulty(event)}>
```

Comment:
https://legacy.reactjs.org/docs/lists-and-keys.html
We can do this with a map?

Reply:
Will refactor in next commit